platform/fetch-node/src/index.ts

```diff
@@ -0,0 +1,20 @@
+import nodeFetch, { RequestInfo, RequestInit, Response } from 'node-fetch'
+import { request as undiciRequest } from 'undici'
+import { setFetchImpl } from '@affinidi/platform-fetch'
+
+type FetchType = (url: RequestInfo, init?: RequestInit) => Promise<Response>
+
+export const useNodeFetch = () => {
+  if (process.env.HTTP_CLIENT === 'undici') {
+    setFetchImpl(async (url, options) => {
+      const response = await undiciRequest((url as URL).href ?? url, options)
+
+      return {
+        status: response.statusCode,
+        json: () => response.body.json(),
```

Comment:
Are we only relying on json? 
Also, no `error` method from original response, or it's enough for our needs?

Reply:
This is the same code that was used for undici before. It's definitely a good idea to improve undici's feature parity with node-fetch, but that appears to be out of scope for this PR.

Reply:
There is a full type safety now (no type assertions in fetch implementations). If whatever current implementation has would not be enough for our needs, we would get a compile-time error.